src/assets/service-worker.ts

```diff
@@ -0,0 +1,32 @@
+/// <reference lib="webworker" />
+declare const self: ServiceWorkerGlobalScope;
+export default null;
+
+import { registerRoute } from 'workbox-routing';
+import { NetworkFirst } from 'workbox-strategies';
+import { CacheableResponsePlugin } from 'workbox-cacheable-response';
+import { ExpirationPlugin } from 'workbox-expiration';
+
+self.addEventListener('install', () => {
+  self.skipWaiting();
+});
+  
+self.addEventListener('activate', (event) => {
+  event.waitUntil(self.clients.claim());
+});
+
+registerRoute(
+  ({ request }) => request.mode === 'navigate',
+  new NetworkFirst({
+    networkTimeoutSeconds: 3,
+    cacheName: 'pages',
+    plugins: [
+      new CacheableResponsePlugin({
+        statuses: [0, 200],
```

Comment:
Should we consider a package like [`http-status-codes`](https://www.npmjs.com/package/http-status-codes) to write `StatusCodes.OK` instead of `200`? Or simply use some `const`s at the top, like `statusOpaque = 0;` and `statusOk = 200`? Or does this only complicate things?

(ironically that package doesn't have the opaque response status; I also had to look that up. So maybe a comment or link on that would be helpful instead)